nexus/db-queries/src/db/datastore/instance.rs

```diff
@@ -1051,6 +1057,16 @@ impl DataStore {
                         .execute_async(&conn)
                         .await?;
 
+                    // Set vCPUs and memory size.
+                    self.instance_set_size_on_conn(
+                        &conn,
+                        &err,
+                        authz_instance,
+                        ncpus,
+                        memory,
+                    )
+                    .await?;
```

Comment:
Calling `instance_set_size_on_conn` will unconditionally check that the instance is in a state in which it can be resized. This means that an `InstanceUpdate` that doesn't change the number of vCPUs or the amount of memory will always check whether the instance is in a resizeable state, and fail if it is not --- meaning that we can no longer set auto-restart policies for running instances. I think we need to check whether the instance's memory/vCPUs are already set to the requested values, and not call this if not.

Reply:
oh yes, that's absolutely right. i've certainly broken that test in the process, even.